cmd/check.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright © 2023 Jacob Aronoff <jacob.aronoff@lightstep.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/util/homedir"
+
+	"github.com/lightstep/collector-cluster-check/pkg/checks"
+	"github.com/lightstep/collector-cluster-check/pkg/checks/certmanager"
+	"github.com/lightstep/collector-cluster-check/pkg/checks/kubernetes"
+	"github.com/lightstep/collector-cluster-check/pkg/checks/lightstep"
+	"github.com/lightstep/collector-cluster-check/pkg/checks/oteloperator"
+	"github.com/lightstep/collector-cluster-check/pkg/checks/prometheus"
+	"github.com/lightstep/collector-cluster-check/pkg/dependencies"
+)
+
+type checkGroup struct {
+	dependencies []dependencies.Initializer
+	checkers     []checks.NewChecker
+}
+
+var (
+	kubeConfig      string
+	accessToken     string
+	http            bool
+	availableChecks = map[string]checkGroup{
+		"metrics": {
```

Comment:
optional (but I think every helpful) -- `grpcurl` (or similar) command to deal with proxy/firewall issues (does _any_ data make it out vs otel-specific data)

Reply:
ah yeah thats like a step between that's useful to have :) I could have a whole DNS check actually